src/ansys/geometry/core/geometry/surfaces/surface.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Provides the ``Surface`` class."""
+from abc import ABC, abstractmethod
+
+from ansys.geometry.core.geometry.parameterization import Parameterization, ParamUV
+from ansys.geometry.core.geometry.surfaces.surface_evaluation import SurfaceEvaluation
+from ansys.geometry.core.math import Matrix44, Point3D
+
+
+class Surface(ABC):
+    """Provides the abstract base class for a 3D surface."""
+
+    @abstractmethod
+    def parameterization(self) -> tuple[Parameterization, Parameterization]:
+        """Parameterize the surface as a tuple (U and V respectively)."""
+        return
+
+    @abstractmethod
+    def contains_param(self, param_uv: ParamUV) -> bool:
+        """Test whether a parameter is within the parametric range of the surface."""
+        return
+
+    @abstractmethod
+    def contains_point(self, point: Point3D) -> bool:
```

Comment:
Proper docstrings... you are missing Parameters and Returns in all these methods

Reply:
The individual parameters and returns are explicitly defined in the classes that implement this abstract class such as cylinder, torus, etc